pallets/phala-world/src/nft_sale.rs

```diff
@@ -0,0 +1,1686 @@
+		pub fn start_incubation(
+			origin: OriginFor<T>,
+			collection_id: CollectionId,
+			nft_id: NftId,
+		) -> DispatchResult {
+			let sender = ensure_signed(origin)?;
+
+			Ok(())
+		}
+
+		/// Feed another origin_of_shell to the current origin_of_shell being incubated. This will
+		/// reduce the time left to incubation if the origin_of_shell is in the top 10 of
+		/// origin_of_shells fed that era.
+		///
+		/// Parameters:
+		/// - origin: The origin of the extrinsic feeding the origin_of_shell
+		/// - collection_id: The collection id of the Origin of Shell RMRK NFT
+		/// - nft_id: The NFT id of the Origin of Shell RMRK NFT
+		#[pallet::weight(10_000 + T::DbWeight::get().reads_writes(1,1))]
+		#[transactional]
+		pub fn feed_origin_of_shell(
+			origin: OriginFor<T>,
+			collection_id: CollectionId,
+			nft_id: NftId,
+		) -> DispatchResult {
+			let sender = ensure_signed(origin)?;
+
+			Ok(())
+		}
+
+		/// Hatch the origin_of_shell that is currently being hatched. This will trigger the end of
+		/// the incubation process and the origin_of_shell will be burned. After burning, the user
+		/// will receive the awakened Shell RMRK NFT
+		///
+		/// Parameters:
+		/// - origin: The origin of the extrinsic incubation the origin_of_shell
+		/// - collection_id: The collection id of the Origin of Shell RMRK NFT
+		/// - nft_id: The NFT id of the Origin of Shell RMRK NFT
+		#[pallet::weight(10_000 + T::DbWeight::get().reads_writes(1,1))]
+		#[transactional]
+		pub fn hatch_origin_of_shell(
+			origin: OriginFor<T>,
+			collection_id: CollectionId,
+			nft_id: NftId,
+		) -> DispatchResult {
+			let sender = ensure_signed(origin)?;
+
+			Ok(())
+		}
+
+		/// This is an admin function to update origin_of_shells incubation times based on being in
+		/// the top 10 of fed origin_of_shells within that era
+		///
+		/// Parameters:
+		/// - origin: The origin of the extrinsic updating the origin_of_shells incubation times
+		/// - collection_id: The collection id of the Origin of Shell RMRK NFT
+		/// - nft_id: The NFT id of the Origin of Shell RMRK NFT
+		/// - reduced_time: The amount of time the origin_of_shell will be reduced by
+		#[pallet::weight(10_000 + T::DbWeight::get().reads_writes(1,1))]
+		#[transactional]
+		pub fn update_incubation_time(
+			origin: OriginFor<T>,
+			collection_id: CollectionId,
+			nft_id: NftId,
+			reduced_time: u64,
+		) -> DispatchResult {
+			// Ensure OverlordOrigin makes call
+			let sender = ensure_signed(origin)?;
+			Self::ensure_overlord(sender)?;
+
+			Ok(())
+		}
+
+		/// Privileged function set the Overlord Admin account of Phala World
+		///
+		/// Parameters:
+		/// - origin: Expected to be called by `OverlordOrigin`
+		/// - new_overlord: T::AccountId
+		#[pallet::weight(0)]
+		pub fn set_overlord(
+			origin: OriginFor<T>,
+			new_overlord: T::AccountId,
+		) -> DispatchResultWithPostInfo {
+			// This is a public call, so we ensure that the origin is some signed account.
+			ensure_root(origin)?;
```

Comment:
Let's make a `EnsureOrigin` type in the `Config` trait, and pass the admin role to this pallet instead of using `ensure_root` directly. We may want to make the `RootOrHalfCouncil` as the admin role

Reply:
Currently we have an `EnsureOrigin` in `Config` trait:
```
/// The origin which may forcibly buy, sell, list/unlist, offer & withdraw offer on Tokens
type OverlordOrigin: EnsureOrigin<Self::Origin>;
```
I can rename it to `RootOrHalfCouncil`

Reply:
OverlordOrigin Trait has been updated and set in runtime to EnsureRootOrHalfCouncil